src/PrestaShopBundle/Translation/View/TreeBuilder.php

```diff
@@ -137,11 +137,12 @@ public function makeTranslationsTree($catalogue)
 
         foreach ($catalogue as $domain => $messages) {
             $tableisedDomain = Inflector::tableize($domain);
-            $parts = array_reverse(explode('_', $tableisedDomain));
+            // the third component of the domain may have underscores, so we need to limit pieces to 3
```

Comment:
can we export this business logic into a specific class to figure out what is done here ?

as a basic rule, everytime I want to comment a block of code, I ask myself: why not write a new function instead with a meaningful name and unit tests ?

This is not naive here, I've already contributed this code and this patch before (but the PR [was not merged](https://github.com/PrestaShop/PrestaShop/pull/13059/files#diff-f22a4655c95d20014e673956d4511214R141)).

Reply:
I think that's a different subject. This fix exists solely to allow underscores in the third part of a translation domain.

Other improvements can be done, for sure, although on a different PR.